benchmarkcommon/src/test/java/com/ss/benchmark/httpclient/HCPerformanceTests.java

```diff
@@ -0,0 +1,399 @@
+
+    @Test(groups = {"sync", "nonblocking"})
+    public void testSyncNonBlockingShortLongPOST() {
+        String method = myName();
+        LOGGER.debug("Start " + method);
+
+        CountDownLatch latch = new CountDownLatch(EXECUTIONS);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            asyncPOST(ECHO_DELAY_LONG_URL,
+                    Payloads.SHORT_JSON,
+                    Payloads.LONG_JSON,
+                    latch,
+                    metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+        }
+
+        LOGGER.debug("Completed " + method);
+    }
+
+    @Test(invocationCount = EXECUTIONS, threadPoolSize = 100, groups = {"async", "nonblocking"})
+    public void testAsyncNonBlockingShortLongPOST() {
+        String method = myName();
+        LOGGER.debug("Start " + method);
+
+        asyncPOST(ECHO_DELAY_LONG_URL,
+                Payloads.SHORT_JSON,
+                Payloads.LONG_JSON,
+                new CountDownLatch(1),
+                metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+
+        LOGGER.debug("Completed " + method);
+    }
+
+    @Test(groups = {"sync", "nonblocking"})
+    public void testSyncNonBlockingLongLongPOST() {
+        String method = myName();
+        LOGGER.debug("Start " + method);
+
+        CountDownLatch latch = new CountDownLatch(EXECUTIONS);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            asyncPOST(ECHO_DELAY_LONG_URL,
+                    Payloads.LONG_JSON,
+                    Payloads.LONG_JSON,
+                    latch,
+                    metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+        }
+
+        LOGGER.debug("Completed " + method);
+    }
+
+    @Test(invocationCount = EXECUTIONS, threadPoolSize = 100, groups = {"async", "nonblocking"})
+    public void testAsyncNonBlockingLongLongPOST() {
+        String method = myName();
+        LOGGER.debug("Start " + method);
+
+        asyncPOST(ECHO_DELAY_LONG_URL,
+                Payloads.LONG_JSON,
+                Payloads.LONG_JSON,
+                new CountDownLatch(1),
+                metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+
+        LOGGER.debug("Completed " + method);
+    }
+
+
+    protected static String myName() {
+        return StackWalker
+                .getInstance(java.lang.StackWalker.Option.RETAIN_CLASS_REFERENCE)
+                .walk(s -> s.skip(1).limit(1).collect(Collectors.toList()))
+                .get(0).getMethodName();
+    }
```

Comment:
I think it would be preferable to not *require* java 11 and we can use testng to supply us with method name with something like
```
    @Test
    public void test(Method m) {
    // m.getName supplies method name. "
    }
```